### DISCUSSION TITLE
Issue de-serializing proto messages on 64 bit machines  for scan_length_prefixed_data() error

### DISCUSSION THREAD

**Discussion by DaemonDave** [protobuf-c/protobuf-c#651] [Q&A]
Here is the proto files causing problems:

point32.proto:

message point32
{
    required float x=1;
    required float y=2;
    required float z=3;
}

header.proto:

message time
{
    required uint32 secs=1;
    required uint32 nanosecs=2;
}
message header 
{
    required uint32 seq=1; 
    required time stamp=2;
    required string frame_id=3;
}

cell.proto:

import "header.proto";
import "point32.proto";
message cellsurface
{
    required uint32 surfaceID=1;
    required float in=2;
    required float out=3;  
    required point32 surf_centre=4;
    optional point32 surf_top_left=5;
    optional point32 surf_top_right=6;
    optional point32 surf_bot_left=7;
    optional point32 surf_bot_right=8;  
}
message cellinfo
{
    required uint32 topic_id=1;
    required uint32 topic_name=2;
    required uint32 topic_msg=3;
    required uint32 message_type=4;
    required uint32 md5sum=5;
    required uint32 buffer_size=6;
    required uint32 seq=7; 
    required time stamp=8;
    required uint32 cellID=10;
    required point32 centre=11;
    required float population=12;  
    required float density=13;  
    required float area=14;  
    required float neighbours=15;  
    required cellsurface surface=16;
}
message cellstats
{
    required uint32 cellID=1;
    required uint32 topic_msg=2;
    required uint32 buffer_size=3;
    required uint32 seq=4; 
    required time stamp=5;
    required point32 centre=6;
    required float population=7;  
    required float density=8;  
    required float area=9;  
    required float neighbours=10;  
    repeated cellsurface surface=11;
}

Now when I make messages longer than say 100 bytes I can't deserialize at the receive end I get an error on ubuntu 15.04

Received: length[169]"error unpacking incoming message"

and on Fedora 21:

Received: length[169]"error unpacking incoming message: "data too short after length-prefix of 29"
 The received length [] is the amount of bytes sent I fprintf to screen.

When I make smaller messages it works fine. When I make them over 100 bytes or use repeated or optional instead of required it faults at receive end. Both machines are 64 bit. 

scan_length_prefixed_data() stops on error

I think there is a bug for 64 bit interpretation inside this function.  

What am I doing wrong with protos or is there a bug in 64 bit versions?

Thanks,
Dave

**Comment by lipnitsk** [comment]
Closing based on the following response on the mailing list from Dave Benson:

> your code has:  
> 
>  // LAST STEP SIZE
>  len = cellinfo__get_packed_size(&combined);
>  combined.buffer_size = len;
>  buf = malloc(len);  
>  cellinfo__pack((const Cellinfo *)&combined,buf);
> 
> but when you change buffer_size, you can affect the packed size.... the cutoff is 127 bytes, which matches your problem...

**Comment by DaemonDave** [comment]
Thank you!
